actnow/accounts/permissions.py

```diff
@@ -1,8 +1,20 @@
 from oauth2_provider.models import get_application_model
-from rest_framework.permissions import BasePermission
+from rest_framework.permissions import SAFE_METHODS, BasePermission
 
 
 class AllowAppicationOwnerOnly(BasePermission):
     def has_permission(self, request, view):
         Application = get_application_model()
         return Application.objects.filter(user=request.user.id).exists()
+
+
+class IsOwnerOrReadOnly(BasePermission):
+    def has_object_permission(self, request, view, obj):
+        if request.method in SAFE_METHODS:
+            return bool(request.user.is_active and request.user.is_authenticated)
+        return bool(obj.id == request.user.id)
```

Comment:
Not sure where this is applicable?

Reply:
While updating an account, we only want the account owner to be able to do edit/delete his/her account but at the same time, other users can view/GET the account.

Reply:
Cool, the owner part is cool but the read only part is where I'm a bit lost: is it read only to authenticated users only or anybody (even non-authenticated users)?

Reply:
>is it read only to authenticated users only or anybody (even non-authenticated users)?

Yes. It's read-only to authenticated users.

Reply:
Cool... so can you know a user is active or not if they're not authenticated?

Reply:
On further testing, any request first checks if a user is active or not. If a user is not active, no further permission checks are performed. So we can update 
`return bool(request.user.is_active and request.user.is_authenticated)`
to 
`return bool(request.user.is_authenticated and not request.user.is_deleted)`

Reply:
Yes we can but my question was if it's necessary: is it possible for `is_authenticated` to be **True** when `is_active` is **False**?

Reply:
no. It's not possible.

Reply:
So we don't need this check, correct?

Reply:
yes. Just `return request.user.is_authenticated` will be enough.